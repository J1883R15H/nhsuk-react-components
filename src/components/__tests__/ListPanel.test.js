import React from 'react';
import { shallow } from 'enzyme';
import ListPanel from '../list-panel';

describe('ListPanel', () => {
  describe('Item', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(<ListPanel.Item>Text</ListPanel.Item>);
      expect(shallowMount).toMatchSnapshot();
      expect(shallowMount.text()).toEqual('Text');
      shallowMount.unmount();
    });
    it('renders href', () => {
      const shallowMount = shallow(
        <ListPanel.Item href="/href">Text</ListPanel.Item>
      );
      expect(
        shallowMount.containsMatchingElement(
          <a className="nhsuk-list-panel__link" href="/href">
            Text
          </a>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
  describe('Panel', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(
        <ListPanel.Panel label="Label">Child</ListPanel.Panel>
      );
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('renders disabled panel', () => {
      const shallowMount = shallow(
        <ListPanel.Panel label="Label" disabled>
          Text
        </ListPanel.Panel>
      );
      expect(
        shallowMount.containsMatchingElement(
          <div className="nhsuk-list-panel__box nhsuk-list-panel__box--with-label">
            <p className="nhsuk-list-panel--results-items__no-results">Text</p>
          </div>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
    it('renders backToTop link', () => {
      const shallowMount = shallow(
        <ListPanel.Panel
          backToTop
          backToTopText="text"
          backToTopHref="#top"
          label="Label"
        >
          Text
        </ListPanel.Panel>
      );
      expect(
        shallowMount.containsMatchingElement(
          <div className="nhsuk-back-to-top">
            <a className="nhsuk-back-to-top__link" href="#top">
              <svg
                className="nhsuk-icon nhsuk-icon__arrow-right"
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 24 24"
                aria-hidden="true"
              >
                <path d="M19.6 11.66l-2.73-3A.51.51 0 0 0 16 9v2H5a1 1 0 0 0 0 2h11v2a.5.5 0 0 0 .32.46.39.39 0 0 0 .18 0 .52.52 0 0 0 .37-.16l2.73-3a.5.5 0 0 0 0-.64z" />
              </svg>
              text
            </a>
          </div>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
  describe('ListPanel', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(<ListPanel>Child</ListPanel>);
      expect(shallowMount).toMatchSnapshot();
      expect(shallowMount.text()).toEqual('Child');
      expect(
        shallowMount.containsMatchingElement(
          <ol className="nhsuk-list">Child</ol>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
});
