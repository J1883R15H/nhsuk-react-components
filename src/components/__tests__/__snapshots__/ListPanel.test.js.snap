// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPanel Item matches snapshot 1`] = `
<li
  className="nhsuk-list-panel__item"
  style={Object {}}
>
  <a
    className="nhsuk-list-panel__link"
    href="#"
  >
    Text
  </a>
</li>
`;

exports[`ListPanel ListPanel matches snapshot 1`] = `
<ol
  className="nhsuk-list"
>
  Child
</ol>
`;

exports[`ListPanel Panel matches snapshot 1`] = `
<div
  className="nhsuk-list-panel"
  style={Object {}}
>
  <h2
    className="nhsuk-list-panel__label"
    id=""
  >
    Label
  </h2>
  <ul
    className="nhsuk-list-panel__list nhsuk-list-panel__list--with-label"
  >
    Child
  </ul>
</div>
`;
